### PR TITLE
Move SiteNoticeAfter hooks from mw-config to MirahezeMagic

### PR DESCRIPTION
**Pull request by Universal-Omega**
The idea behind this is to later allow for localisation of the notices. This changes from the mw-config hook also. In this, I will not display notices for both closed and inactive, only closed or inactive. That can be changed it wanted to preserve former behavior.

### CODE REVIEW

**Comment by JohnFLewis** [comment]
If we’re moving this to an extension, we can use RemoteWiki to check if a wiki is active/inactive. The $cw* state variables were added just to allow this to be done in LS

File: includes/MirahezeMagicHooks.php [hunk 1]
```diff
@@ -277,4 +277,31 @@ public static function onUserGetRightsRemove( User $user, array &$aRights ) {
 			}
 		}
 	}
+
+	public static function onSiteNoticeAfter( &$siteNotice, $skin ) {
+		$config = MediaWikiServices::getInstance()->getConfigFactory()->makeConfig( 'mirahezemagic' );
+		$wiki = new RemoteWiki( $config->get( 'DBname' ) );
```
Comment (paladox) [inline includes/MirahezeMagicHooks.php:283]: I think using this would be a performance issue? It would run on every request.

Comment (Universal-Omega) [reply includes/MirahezeMagicHooks.php:283]: I don't think it'd be much of a performance issue tbh, it's done like this in many other places throughout, although maybe it's not run on every page load there, so should it be converted back to `$cw*` variables then?

Comment (paladox) [reply includes/MirahezeMagicHooks.php:283]: yes, I think we should use the variable.

Comment (paladox) [reply includes/MirahezeMagicHooks.php:283]: John told me he suggested something so you should follow what he says.

Comment (Universal-Omega) [reply includes/MirahezeMagicHooks.php:283]: Yes, the way it is now is what I thought John was suggesting, unless I did it wrong/there is another way?

Comment (paladox) [reply includes/MirahezeMagicHooks.php:283]: Nope, this is fine. John says there wont be any performance issues.

Comment (Universal-Omega) [reply includes/MirahezeMagicHooks.php:283]: Thanks!

**Review by Reception123 — APPROVED** [review]

**Pull request merged**